### PR TITLE
Add turbo button

### PR DESCRIPTION
**Pull request by IgorMinar**
//cc: @alexeagle @tbosch

### CODE REVIEW

**Comment by IgorMinar** [comment]
I haven't adjusted the node and dart trees yet. Let's see if this passes on CI and if it does let's merge it. I'll redo the other two trees tomorrow in as followup PR.

note that I called the flag `--projects`. this is because I want to rename modules to projects (or just wait until modules goes away when @yjbanov's directory layout is implemented).

**Comment by IgorMinar** [comment]
We can improve the perf further by dropping the ES6 emit when `--projects` is specified (or removing it completely as proposed by @alexeagle)

File: gulpfile.js [hunk 1]
```diff
@@ -46,10 +46,36 @@ var dartSdk = require('./tools/build/dart');
...
+                            map[projectName] = true;
+                            return map;
+                          }, {});
+
+function printModulesWarning() {
+  if (!cliArgs.projects && !process.env.CI) {
+    // if users didn't specify projects to build, tell them why and how they should
+    console.warn(
+        "Pro Tip: Did you know that you can speed up your build by specifying project name(s)?");
+    console.warn("         It's like pressing the turbo button in the old days, but better!");
```
Comment (alexeagle) [inline gulpfile.js]: ಠ_ಠ

**Comment by IgorMinar** [comment]
for casual testing it looks like `--projects=angular2` speeds up the build as follows:

|                 | build time | test time |
-------------- | --------------|-------------
| **first build**  | 21s -> 17s | 14s -> 12s |
| **rebuild**     | 4.5s -> 3.8s | 11s -> 10s |


We are still spending too much time in Funnels, so there is more we can do to optimize this.

Before:

```
Slowest Trees                                 | Total               
----------------------------------------------+---------------------
DiffingTSCompiler                             | 2285ms              
Funnel                                        | 753ms               
DiffingTSCompiler                             | 321ms               

Slowest Trees (cumulative)                    | Total (avg)         
----------------------------------------------+---------------------
DiffingTSCompiler (2)                         | 2606ms (1303 ms)    
Funnel (85)                                   | 1349ms (15 ms)      
MergeTrees (5)                                | 393ms (78 ms)       
```


After:

```
Slowest Trees                                 | Total               
----------------------------------------------+---------------------
DiffingTSCompiler                             | 1977ms              
Funnel                                        | 846ms               

Slowest Trees (cumulative)                    | Total (avg)         
----------------------------------------------+---------------------
DiffingTSCompiler (2)                         | 2092ms (1046 ms)    
Funnel (6)                                    | 1042ms (173 ms)     
MergeTrees (5)                                | 376ms (75 ms) 
```


Dropping the ES6 output is likely the most impactful thing we can do next.

**Pull request closed without merging**